### ISSUE TITLE
Hostname of job runner does not match for

### ISSUE THREAD

**Issue by karakanb** [airflow-helm/charts#596]
### Checks

- [X] I have checked for [existing issues](https://github.com/airflow-helm/charts/issues).
- [X] This report is about the [`User-Community Airflow Helm Chart`](https://github.com/airflow-helm/charts/tree/main/charts/airflow).

### Chart Version

8.6.0

### Kubernetes Version

```shell
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.6", GitCommit:"ad3338546da947756e8a88aa6822e9c11e7eac22", GitTreeState:"clean", BuildDate:"2022-04-14T08:41:58Z", GoVersion:"go1.18.1", Compiler:"gc", Platform:"darwin/arm64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.11", GitCommit:"38d3c1f3d5306401bcf39a71bad3b5a5106033d7", GitTreeState:"clean", BuildDate:"2022-03-16T14:02:06Z", GoVersion:"go1.16.15", Compiler:"gc", Platform:"linux/amd64"}
```

### Helm Version

```shell
version.BuildInfo{Version:"v3.8.2", GitCommit:"6e3701edea09e5d55a8ca2aae03a68917630e91b", GitTreeState:"clean", GoVersion:"go1.18.1"}
```

### Description

Some of my long-running tasks are randomly failing with the following error:

```
2022-06-03 01:34:47,118: ERROR/ForkPoolWorker-189] Failed to execute task Hostname of job runner does not match.
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/executors/celery_executor.py", line 121, in _execute_in_fork
    args.func(args)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/cli_parser.py", line 48, in command
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/cli.py", line 92, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 298, in task_run
    _run_task_by_selected_method(args, dag, ti)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 105, in _run_task_by_selected_method
    _run_task_by_local_task_job(args, ti)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 163, in _run_task_by_local_task_job
    run_job.run()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/jobs/base_job.py", line 246, in run
    self._execute()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/jobs/local_task_job.py", line 134, in _execute
    self.heartbeat()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/jobs/base_job.py", line 227, in heartbeat
    self.heartbeat_callback(session=session)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 67, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/jobs/local_task_job.py", line 196, in heartbeat_callback
    raise AirflowException("Hostname of job runner does not match")
airflow.exceptions.AirflowException: Hostname of job runner does not match
2022-06-03 01:34:47,154: ERROR/ForkPoolWorker-189] Task airflow.executors.celery_executor.execute_command[75a5b198-606e-4625-a1c2-eb14a150b445] raised unexpected: AirflowException('Celery command failed on host: airflow-v2-worker-1.airflow-v2-worker.airflow.svc.cluster.local')
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/celery/app/trace.py", line 451, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/celery/app/trace.py", line 734, in __protected_call__
    return self.run(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/executors/celery_executor.py", line 90, in execute_command
    _execute_in_fork(command_to_exec, celery_task_id)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/executors/celery_executor.py", line 101, in _execute_in_fork
    raise AirflowException('Celery command failed on host: ' + get_hostname())
```

### Relevant Logs

These are the logs from the worker during the failure:

```shell
2022-06-03 01:34:47,118: ERROR/ForkPoolWorker-189] Failed to execute task Hostname of job runner does not match.
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/executors/celery_executor.py", line 121, in _execute_in_fork
    args.func(args)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/cli_parser.py", line 48, in command
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/cli.py", line 92, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 298, in task_run
    _run_task_by_selected_method(args, dag, ti)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 105, in _run_task_by_selected_method
    _run_task_by_local_task_job(args, ti)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/cli/commands/task_command.py", line 163, in _run_task_by_local_task_job
    run_job.run()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/jobs/base_job.py", line 246, in run
    self._execute()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/jobs/local_task_job.py", line 134, in _execute
    self.heartbeat()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/jobs/base_job.py", line 227, in heartbeat
    self.heartbeat_callback(session=session)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/utils/session.py", line 67, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/jobs/local_task_job.py", line 196, in heartbeat_callback
    raise AirflowException("Hostname of job runner does not match")
airflow.exceptions.AirflowException: Hostname of job runner does not match
2022-06-03 01:34:47,154: ERROR/ForkPoolWorker-189] Task airflow.executors.celery_executor.execute_command[75a5b198-606e-4625-a1c2-eb14a150b445] raised unexpected: AirflowException('Celery command failed on host: airflow-v2-worker-1.airflow-v2-worker.airflow.svc.cluster.local')
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/celery/app/trace.py", line 451, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/celery/app/trace.py", line 734, in __protected_call__
    return self.run(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/executors/celery_executor.py", line 90, in execute_command
    _execute_in_fork(command_to_exec, celery_task_id)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/executors/celery_executor.py", line 101, in _execute_in_fork
    raise AirflowException('Celery command failed on host: ' + get_hostname())
```

### Custom Helm Values

```yaml
airflow:
    fernetKey: "some-key"
    webserverSecretKey: "some-other-key"
    executor: CeleryKubernetesExecutor
    users: []
    image:
        repository: hub.docker.com/my-custom-image
        tag: latest
        pullSecret: my-pull-secret

    config:
        # Security
        AIRFLOW__API__AUTH_BACKEND: "airflow.api.auth.backend.deny_all"

        AIRFLOW__KUBERNETES__LOGS_VOLUME_CLAIM: "airflow-v2-logs"
        AIRFLOW__KUBERNETES__DAGS_VOLUME_SUBPATH: "repo"
        AIRFLOW__KUBERNETES__DELETE_WORKER_PODS: "True"
        AIRFLOW__KUBERNETES__WORKER_PODS_CREATION_BATCH_SIZE: "20"

        # DAGS
        AIRFLOW__SCHEDULER__DAG_DIR_LIST_INTERVAL: "120"
        AIRFLOW__SCHEDULER__MIN_FILE_PROCESS_INTERVAL: "30"

        # Webserver
        AIRFLOW__WEBSERVER__BASE_URL: "https://my-domain.com"
        AIRFLOW__WEBSERVER__WORKER_CLASS: "gevent"
        AIRFLOW__WEBSERVER__EXPOSE_CONFIG: "False"
        AIRFLOW__WEBSERVER__RBAC: "True"
        AIRFLOW__WEBSERVER__SECRET_KEY: "some-secret"

        # Core
        AIRFLOW__CORE__SECURE_MODE: "True"
        AIRFLOW__CORE__LOAD_EXAMPLES: "False"
        AIRFLOW__CORE__PARALLELISM: "30"
        AIRFLOW__CORE__MAX_ACTIVE_TASKS_PER_DAG: "15"
        AIRFLOW__CORE__MAX_ACTIVE_RUNS_PER_DAG: "1"

        AIRFLOW__CELERY_BROKER_TRANSPORT_OPTIONS__VISIBILITY_TIMEOUT: "64800"
        AIRFLOW__CELERY__WORKER_AUTOSCALE: "8,4"

        ## Email (SMTP)
        AIRFLOW__EMAIL__EMAIL_BACKEND: "airflow.utils.email.send_email_smtp"
        AIRFLOW__SMTP__SMTP_HOST: "smtpmail.example.com"
        AIRFLOW__SMTP__SMTP_STARTTLS: "False"
        AIRFLOW__SMTP__SMTP_SSL: "False"
        AIRFLOW__SMTP__SMTP_PORT: "25"
        AIRFLOW__SMTP__SMTP_MAIL_FROM: "admin@example.com"

        ## Disable noisy "Handling signal: ttou" Gunicorn log messages
        GUNICORN_CMD_ARGS: "--log-level WARNING"

    kubernetesPodTemplate:
        resources:
            requests:
                cpu: "200m"
                memory: "250Mi"
            limits:
                cpu: "400m"
                memory: "2Gi"

    dbMigrations:
        runAsJob: true

    localSettings:
        stringOverride: |
            STATE_COLORS = {
                "queued": "gray",
                "running": "lime",
                "success": "#2596be",
                "failed": "red",
                "up_for_retry": "gold",
                "up_for_reschedule": "turquoise",
                "upstream_failed": "orange",
                "skipped": "pink",
                "scheduled": "tan",
                "deferred": "mediumpurple",
            }


logs:
    path: /opt/airflow/logs
    persistence:
        enabled: true
        storageClass: nfs

workers:
    enabled: true
    replicas: 5
    logCleanup:
        enabled: false

    podDisruptionBudget:
        enabled: true
        minAvailable: 1

    resources:
        requests:
            cpu: "500m"
            memory: "2Gi"
        limits:
            memory: "4Gi"

    autoscaling:
        enabled: true
        maxReplicas: 6
        metrics:
            - type: Resource
              resource:
                  name: memory
                  target:
                      type: Utilization
                      averageUtilization: 70
    
    celery:
        gracefullTermination: true
        gracefullTerminationPeriod: 600


flower:
    enabled: true
    replicas: 1
    resources: {}
    service:
        type: ClusterIP
        externalPort: 5555
redis:
    enabled: false

externalRedis:
    host: my-redis-host
    port: 12345
    username: some-username
    passwordSecret: "redis-secret"
    passwordSecretKey: "redis-password"
    properties: "?ssl_cert_reqs=CERT_OPTIONAL"

dags:
    path: /opt/airflow/dags
    gitSync:
        enabled: true
        repo: "some-repo"
        sshSecret: "airflow-ssh"
        sshSecretKey: "ssh-privatekey"
        depth: 1
        resources:
            requests:
                cpu: "50m"
                memory: "50Mi"
            limits:
                memory: "250Mi"

web:
    replicas: 2
    resources:
        requests:
            cpu: "1"
            memory: "4Gi"
        limits:
            memory: "4Gi"
    podDisruptionBudget:
        enabled: true
        minAvailable: 1

scheduler:
    replicas: 2
    logCleanup:
        enabled: false
    resources:
        requests:
            cpu: "1"
            memory: "2Gi"
        limits:
            memory: "3Gi"

triggerer:
    enabled: false

ingress:
    enabled: true
    web:
        host: "my-domain.com"
        annotations:
            nginx.ingress.kubernetes.io/affinity: cookie

postgresql:
    enabled: false

externalDatabase:
    type: postgres
    host: my-db
    port: 12345
    database: db-name
    user: user
    passwordSecret: "airflow-postgres"
    passwordSecretKey: "password"
    properties: "?sslmode=require"

serviceMonitor:
    enabled: false

prometheusRule:
    enabled: false

pgbouncer:
    enabled: false
```

**Comment by thesuperzapper** [comment]
@karakanb what version of Airflow are you using?

This might be related to some kind of intermittent DNS failure in your Kubernetes cluster, are you able to check the logs for your cluster DNS?

Also, I found that some people fixed this by setting `AIRFLOW__CORE__HOSTNAME_CALLABLE` to `socket.gethostname` ([StackOverflow link](https://stackoverflow.com/questions/57681573/how-to-fix-the-error-airflowexceptionhostname-of-job-runner-does-not-match)), but I am not sure if that will fix your case.

**Comment by karakanb** [comment]
I have received this error with v2.2.5, and apparently that callable name has been fixed after v2.1. Unfortunately I don't have the logs anymore, and I haven't observed this issue for the past 2 days.

I'll close this issue for now. I have just updated to v2.3.2, so I'll observe for a few more days, and if I encounter again I'll reopen the issue with hopefully more logs. Thanks for looking into it. 🙂
